json_encoder_test.go

```diff
@@ -126,49 +152,51 @@ func TestJSONAddFloat64(t *testing.T) {
 }
 
 func TestJSONWriteEntry(t *testing.T) {
-	withJSONEncoder(func(enc *jsonEncoder) {
-		sink := bytes.NewBuffer(nil)
-		entry := &Entry{Level: InfoLevel, Message: `hello\`, Time: time.Unix(0, 0)}
-
-		assert.Equal(t, errNilSink, enc.WriteEntry(
-			nil,
-			entry.Message,
-			entry.Level,
-			entry.Time,
-		), "Expected an error writing to a nil sink.")
-
-		// Messages should be escaped.
-		err := enc.WriteEntry(sink, entry.Message, entry.Level, entry.Time)
-		assert.NoError(t, err, "WriteEntry returned an unexpected error.")
-		assert.Equal(t,
-			`{"msg":"hello\\","level":"info","ts":0,"fields":{"foo":"bar"}}`,
-			strings.TrimRight(sink.String(), "\n"),
-		)
+	sink := newTestBuffer()
+	entry := &Entry{Level: InfoLevel, Message: `hello\`, Time: time.Unix(0, 0)}
+	enc := newJSONEncoder()
+	enc.AddString("foo", "bar")
 
-		// We should be able to re-use the encoder, preserving the accumulated
-		// fields.
-		sink.Reset()
-		err = enc.WriteEntry(sink, entry.Message, entry.Level, time.Unix(100, 0))
-		assert.NoError(t, err, "WriteEntry returned an unexpected error.")
-		assert.Equal(t,
-			`{"msg":"hello\\","level":"info","ts":100,"fields":{"foo":"bar"}}`,
-			strings.TrimRight(sink.String(), "\n"),
-		)
-	})
+	assert.Equal(t, errNilSink, enc.WriteEntry(
+		nil,
+		entry.Message,
+		entry.Level,
+		entry.Time,
+	), "Expected an error writing to a nil sink.")
+
+	// Messages should be escaped.
+	err := enc.WriteEntry(sink, entry.Message, entry.Level, entry.Time)
+	assert.NoError(t, err, "WriteEntry returned an unexpected error.")
+	assert.Equal(
+		t,
+		`{"foo":"bar","msg":"hello\\","level":"info","ts":0}`,
+		sink.FirstLine(),
```

Comment:
Instead of `FirstLine`, can we have a method that returns the contents with the new line stripped? We want the tests to fail if there is an unexpected extra line, the easiest way seems to be to keep the previous logic, but have it as a method on `testBuffer`